articles/howtos/create-new-decorator.md

```diff
@@ -0,0 +1,222 @@
+
+![Startup](images/create-new-decorator/MyDecoratorModuleStartup.png)
+
+Go ahead and create your Application in Intent Architect that will be used to create the Module that will contain the new Decorator for the `StartupTemplate`.
+
+Once the generation is done, create a new Decorator called `StaticFileServerDecorator`.
+
+![New Decroator](images/create-new-decorator/NewDecorator.png)
+
+We need to specify the Contract name for that Decorator so that it will know for which Template it is meant to Decorate.
+
+To recap, the Template we're trying to Decorate is the `StartupTemplate` and it requires a Decorator that adheres to this interface:
+
+```csharp
+public interface IStartupTemplateContract : ITemplateDecorator
+{
+    string ConfigureCode();
+}
+```
+
+It has a Full Namespace of: `MyCompany.MyModule.Templates.StartupTemplate.IStartupTemplateContract`
+
+So that is the value that we need to specify in the `Full TypeName` field for that Decorator.
+
+Also, click on the `Declare Usings` flag in the `Decorator Settings` panel.
+
+![Add Type FullName](images/create-new-decorator/AddFQDN.png)
+
+Run the code generation.
+
+![Code Generation Staging](images/create-new-decorator/CodeGenerationStaging.png)
+
+Open Visual Studio and navigate to where the code has been generated and open up the `MyCompany.MyDecoratorModule` solution.
+
+You will notice that it has generated a Decorator as well as a Registration for the Decorator.
+
+![Solution Tree Structure](images/create-new-decorator/SolutionTreeStructure.png)
+
+Although, you will notice that the IDE reports a problem with our namespace we gave it.
+
+![IDE Error](images/create-new-decorator/IDEError.png)
+
+## Ensure the "blueprint" (or contract) assembly is referenced
+
+To fix this issue, we need to make sure that we reference the assembly that contains `IStartupTemplateContract`.
+
+In our case this is simple. We add a project reference to that solution.
+
+![Add Existing Project](images/create-new-decorator/AddExistingProject.png)
+
+Let's add that module as an existing project into our current solution.
+
+![Add Project Reference](images/create-new-decorator/AddProjectReference.png)
+
+Now we add that project as a reference to our new module project.
+
+## Implement the Decorator
+
+This should resolve our initial problem which is that the interface was not found but now we need to implement that interface, so we're not out of the woods yet.
+
+![Go To Implement Interface Command](images/create-new-decorator/GoToImplementInterfaceCommand.png)
+
+Once we instructed Visual Studio to implement this interface, you will notice that it added two things:
+
+```csharp
+public int Priority => throw new NotImplementedException();
+
+public string ConfigureCode()
+{
+    throw new NotImplementedException();
```

Comment:
```suggestion
    return @"app.UseStaticFiles(new StaticFileOptions
    {
        FileProvider = new PhysicalFileProvider(
            Path.Combine(Directory.GetCurrentDirectory(), ""MyStaticFiles"")),
        RequestPath = ""/StaticFiles""
    });";
}
```

Reply:
Already explained above.

Reply:
Sorry, I'm not sure which comment you're referring to now, and I've since added [this comment](https://github.com/IntentSoftware/Docs/pull/1/files#r366506549).

How do you feel now about the suggested change?